cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+	log "github.com/sirupsen/logrus"
+)
+
+// CloudWatchLogs is the internal cloudwatch logsobject which holds session
+// and configuration information
+type CloudWatchLogs struct {
+	Service cloudwatchlogsiface.CloudWatchLogsAPI
+}
+
+// Event is a cloudwatchlogs Event
+type Event struct {
+	Message   string
+	Timestamp int64
+}
+
+// NewSession builds a new aws cloudwatchlogs session
+func NewSession(region, akid, secret string) CloudWatchLogs {
+	c := CloudWatchLogs{}
+	log.Infof("Creating new session with key id %s in region %s", akid, region)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(akid, secret, ""),
+		Region:      aws.String(region),
+	}))
+	c.Service = cloudwatchlogs.New(sess)
+	return c
+}
+
+func (c *CloudWatchLogs) GetLogEvents(ctx context.Context, input *cloudwatchlogs.GetLogEventsInput) (*cloudwatchlogs.GetLogEventsOutput, error) {
+	if input == nil {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	output, err := c.Service.GetLogEventsWithContext(ctx, input)
+	if err != nil {
+		return nil, ErrCode("failed to get log events", err)
+	}
+
+	return output, nil
+}
+
+// CreateLogGroup creates a cloudwatchlogs log group
+func (c *CloudWatchLogs) CreateLogGroup(ctx context.Context, group string, tags map[string]*string) error {
+	if group == "" {
+		return apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	if _, err := c.Service.CreateLogGroupWithContext(ctx, &cloudwatchlogs.CreateLogGroupInput{
+		LogGroupName: aws.String(group),
+		Tags:         tags,
+	}); err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case cloudwatchlogs.ErrCodeResourceAlreadyExistsException:
+				log.Warnf("cloudwatch log group (%s) already exists, continuing: (%s)", group, err)
+			default:
+				msg := fmt.Sprintf("failed to create log group (%s)", group)
+				return ErrCode(msg, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// UpdateRetention changes the retention (in days) for logs in a log group
+func (c *CloudWatchLogs) UpdateRetention(ctx context.Context, group string, retention int64) error {
+	if group == "" {
+		return apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	_, err := c.Service.PutRetentionPolicyWithContext(ctx, &cloudwatchlogs.PutRetentionPolicyInput{
+		LogGroupName:    aws.String(group),
+		RetentionInDays: aws.Int64(retention),
+	})
+	if err != nil {
+		msg := fmt.Sprintf("failed to update retention policy for log group (%s)", group)
+		return ErrCode(msg, err)
+	}
+
+	return nil
+}
+
+// CreateLogStream creates a cloudwatchlogs log stream
+func (c *CloudWatchLogs) CreateLogStream(ctx context.Context, group, stream string) error {
```

Comment:
group does not have a type

Reply:
it does - it's `string`.  you can omit the type for a list of variables of the same type

Reply:
I see.